### ISSUE TITLE
tf.keras.layers.ZeroPadding2D crashes (segfault)

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#47289]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
tf.keras.layers.ZeroPadding2D crashes (segfault) when `padding` is large and input contains 0 in shape.

**Describe the expected behavior**
Expect graceful exception messages instead of crash

**Standalone code to reproduce the issue**
~~~python
import numpy as np
import tensorflow as tf
layer = tf.keras.layers.ZeroPadding2D(padding=420958214)
layer(np.ones((0, 4, 4, 4)))
~~~

Output
~~~python
Segmentation fault (core dumped)
~~~

**Comment by ravikyram** [comment]
Was able to reproduce the issue with TF v2.1, TF v2.4 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/ravikyram/5a7bf0505aeefa574e68412d2f18a32e/untitled676.ipynb). Thanks!

**Comment by sachinprasadhs** [comment]
Was able to run your code without any errors in Tensorflow version 2.5, please find the gist [here](https://colab.research.google.com/gist/googly789/bb8e74e4bf27cc8440665f94181eb0bc/untitled96.ipynb).

**Comment by sushreebarsa** [comment]
Closing this issue as it is fixed in latest version of TensorFlow. Please feel free to reopen the issue if you still have a concern. Thanks!
